### ISSUE TITLE
Hydrogen bond analysis with PBC - distance criteria

### ISSUE THREAD

**Issue by conicd** [MDAnalysis/mdanalysis#1878]
Hi guys,

I have very recently started analyzing my system with MDAnalysis and am facing following issue:

### Expected behaviour
Obtaining a list of hydrogen bonds including the bonds through periodic boundaries,
utilizing hbond_analysis.py in mda-0.18.0

### Actual behaviour
Obtained incomplete list of hydrogen bonds for predefined parameters.
Hydrogen bonds that are missing are PBC criteria related
(concluded from visual inspection of the trajectory snapshots).

Comment:
Seems like that, even though significant progress is made in #1782 in terms of PBC,
distance criteria for hbond_analysis with PBC might not be met.

For instance, in hbond_analysis.py, following piece of the code

1049     @staticmethod
1050     def calc_eucl_distance(a1, a2):
1051         """Calculate the Euclidean distance between two atoms. """
1052         return norm(a2.position - a1.position)

is used for the distance criteria, which seems not to take into account PBC.

Hope that you will perceive the comment as an attempt of mine
to understand the code (and coding) since I am new in this field
and am aware that the comment might be wrong.

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
import MDAnalysis.analysis.hbonds as hbnd
import pandas as pd

mda.core.flags['use_periodic_selections'] = True

u =mda.Universe(psf,dcd)

sikeg=u.select_atoms("segid HETA")
sugar=u.select_atoms("segid HETB")
h_acceptor_list=list(sikeg[0:35].names)+list(sikeg[40:-1].names)

class HydrogenBondAnalysis_OtherFF(hbnd.HydrogenBondAnalysis):
    DEFAULT_DONORS = {"OtherFF": tuple(set(["O2","O4","O7","O8","O9","N"]))}
    DEFAULT_ACCEPTORS = {"OtherFF": tuple(set(h_acceptor_list))}

h=HydrogenBondAnalysis_OtherFF(u,selection1="segid HETB",selection2="segid HETA",\
selection1_type="donor",start=50000,step=2,stop=60000,distance=3.5,distance_type="heavy",\
angle=150.0,forcefield="OtherFF",pbc=True)

h.run()
h.generate_table()
df = pd.DataFrame.from_records(h.table)
with open("testfile2.txt","w") as outfile:
    df.to_string(outfile)

```

### Currently version of MDAnalysis:
0.18.0

**Comment by orbeckst** [comment]
@richardjgowers did the fix in PR #1879 not fix everything?

**Comment by richardjgowers** [comment]
@orbeckst I just wanted to double check with the reporter that it fixed their issue, but it's probably fixed now yeah
